src/app/pages/onboarding/set-password/components/password-field.utils.ts

```diff
@@ -0,0 +1,32 @@
+import { color } from '@stacks/ui';
+
+import {
+  averageStrengthScore,
+  poorStrengthScore,
+  strongStrengthScore,
+  ValidatedPassword,
+  weakStrengthScore,
+} from '@app/common/validation/validate-password';
+
+export const defaultColor = '#EFEFF2';
+
+enum PasswordStrength {
+  Poor = 'Poor',
+  Weak = 'Weak',
+  Average = 'Average',
+  Strong = 'Strong',
+}
+
+export function getIndicatorsOfPasswordStrength(result: ValidatedPassword) {
+  if (result.score === poorStrengthScore) {
+    return { strengthColor: color('feedback-error'), strengthText: PasswordStrength.Poor };
+  } else if (result.score === weakStrengthScore) {
+    return { strengthColor: color('feedback-alert'), strengthText: PasswordStrength.Weak };
+  } else if (result.score === averageStrengthScore) {
+    return { strengthColor: '#F0C000', strengthText: PasswordStrength.Average };
+  } else if (result.score === strongStrengthScore || result.meetsAllStrengthRequirements) {
+    return { strengthColor: color('feedback-success'), strengthText: PasswordStrength.Strong };
+  } else {
+    return { strengthColor: defaultColor, strengthText: '-' };
+  }
```

Comment:
Imo this'd better use either, a switch statement, or an object that you access by property like

```ts
const strengthStyles = {
  [PasswordStrength.Poor]: { strengthColor: color('feedback-error'), strengthText: PasswordStrength.Poor },
  ...
}
export function getIndicatorsOfPasswordStrength(result: ValidatedPassword) {
  return strengthStyles[result.score];
}
```
Not that there's anything wrong with `else if` statements, but I almost never use them as I find there are more readable ways to express the same.

Reply:
Ok, yep, I can change it.